File: backend/app/schemas/football_national_team.py
from pydantic import BaseModel
from datetime import date, datetime
from typing import Optional, List, Dict

class FootballNationalTeamBase(BaseModel):
    name: str
    country: Optional[str] = None
    confederation: Optional[str] = None
    founded_year: Optional[int] = None
    stadium: Optional[str] = None
    manager: Optional[str] = None
    nickname: Optional[str] = None
    image_url: Optional[str] = None
    website: Optional[str] = None
    description: Optional[str] = None
    ranking: Optional[int] = None
    category: Optional[str] = "men"
    
    # Enhanced Statistics
    total_trophies: Optional[int] = 0
    world_cup_titles: Optional[int] = 0
    captain: Optional[str] = None
    main_rivals: Optional[str] = None
    
    # Detailed Honors
    honors_json: Optional[Dict[str, int]] = None

class FootballNationalTeamCreate(FootballNationalTeamBase):
    pass

class FootballNationalTeamUpdate(FootballNationalTeamBase):
    name: Optional[str] = None

class FootballRankingHistoryPoint(BaseModel):
    ranking: int
    date: date

class FootballNationalTeam(FootballNationalTeamBase):
    id: int
    last_updated: Optional[datetime] = None
    ranking_history: Optional[List[FootballRankingHistoryPoint]] = None
    highest_ranking: Optional[int] = None
    highest_ranking_date: Optional[date] = None
    career_high_rank: Optional[int] = None
    career_high_date: Optional[date] = None

    class Config:
        from_attributes = True

class FootballNationalTeamList(BaseModel):
    items: List[FootballNationalTeam]
    total: int
    page: int
    size: int

File: frontend/lib/models/football_national_team.dart
import '../widgets/ranking_graph.dart';

class FootballNationalTeam {
  final int id;
  final String name;
  final String? country;
  final String? confederation;
  final int? foundedYear;
  final String? stadium;
  final String? manager;
  final String? nickname;
  final String? imageUrl;
  final String? website;
  final String? description;
  final int? ranking;
  final String category;
  final List<RankingPoint>? rankingHistory;
  final int? highestRanking;
  final DateTime? highestRankingDate;

  // Enhanced Statistics
  final int totalTrophies;
  final int worldCupTitles;
  final String? captain;
  final String? mainRivals;

  // Detailed Honors
  final Map<String, int>? honors;

  FootballNationalTeam({
    required this.id,
    required this.name,
    this.country,
    this.confederation,
    this.foundedYear,
    this.stadium,
    this.manager,
    this.nickname,
    this.imageUrl,
    this.website,
    this.description,
    this.ranking,
    required this.category,
    this.rankingHistory,
    this.highestRanking,
    this.highestRankingDate,
    this.totalTrophies = 0,
    this.worldCupTitles = 0,
    this.captain,
    this.mainRivals,
    this.honors,
  });

  factory FootballNationalTeam.fromJson(Map<String, dynamic> json) {
    return FootballNationalTeam(
      id: json['id'],
      name: json['name'],
      country: json['country'],
      confederation: json['confederation'],
      foundedYear: json['founded_year'],
      stadium: json['stadium'],
      manager: json['manager'],
      nickname: json['nickname'],
      imageUrl: json['image_url'],
      website: json['website'],
      description: json['description'],
      ranking: json['ranking'],
      category: json['category'] ?? 'men',
      rankingHistory: json['ranking_history'] != null
          ? (json['ranking_history'] as List)
              .map((item) => RankingPoint(
                    ranking: item['ranking'],
                    date: DateTime.parse(item['date']),
                  ))
              .toList()
          : null,
      highestRanking: json['highest_ranking'] ?? json['career_high_rank'],
      highestRankingDate: json['highest_ranking_date'] != null
          ? DateTime.parse(json['highest_ranking_date'])
          : (json['career_high_date'] != null
              ? DateTime.parse(json['career_high_date'])
              : null),
      totalTrophies: json['total_trophies'] ?? 0,
      worldCupTitles: json['world_cup_titles'] ?? 0,
      captain: json['captain'],
      mainRivals: json['main_rivals'],
      honors: json['honors_json'] != null
          ? Map<String, int>.from(json['honors_json'])
          : null,
    );
  }
}

class FootballNationalTeamListResponse {
  final List<FootballNationalTeam> items;
  final int total;
  final int page;
  final int size;

  FootballNationalTeamListResponse({
    required this.items,
    required this.total,
    required this.page,
    required this.size,
  });

  factory FootballNationalTeamListResponse.fromJson(Map<String, dynamic> json) {
    return FootballNationalTeamListResponse(
      items: (json['items'] as List)
          .map((i) => FootballNationalTeam.fromJson(i))
          .toList(),
      total: json['total'],
      page: json['page'],
      size: json['size'],
    );
  }
}
